### PR TITLE
test: block filter better coverage

### PR DESCRIPTION
**Pull request by szuecs**
test: block filter better coverage

Signed-off-by: Sandor Szuecs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: filters/block/block_test.go [hunk 1]
```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
...
+	t.Run("maxBuffer read on closed reader", func(t *testing.T) {
+		pipeR, pipeW := io.Pipe()
+		initialContent := []byte("fppppppppp")
+		go pipeW.Write(initialContent)
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(pipeR, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(initialContent)+10)
+		pipeR.Close()
+		_, err := bmb.Read(p)
+		if err == nil || err.Error() != "io: read/write on closed pipe" {
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:117]: There is [io.ErrClosedPipe](https://pkg.go.dev/io#pkg-variables) that may be used here

File: filters/block/block_test.go [hunk 2]
```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
...
+	t.Run("maxBuffer read on initial closed reader", func(t *testing.T) {
+		pipeR, _ := io.Pipe()
+		initialContent := []byte("fppppppppp")
+		toblockList := []toblockKeys{{str: []byte(".class")}}
+		bmb := newMatcher(pipeR, toblockList, 5, maxBufferBestEffort)
+		p := make([]byte, len(initialContent)+10)
+		pipeR.Close()
+		bmb.Close()
+		_, err := bmb.Read(p)
+		if err == nil || err.Error() != "reader closed" {
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:131]: Error string matching is a good fit for `assert.EqualError` :grin:

File: filters/block/block_test.go [hunk 3]
```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
...
+			t.Error("CreateFilter with non string args should fail")
+		}
+	})
+}
+
+func TestBlock(t *testing.T) {
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte("OK"))
+	}))
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:159]: I believe you'd need to `defer backend.Close()`. Maybe you could add main_test.go like we did for other packages to catch this in the future

File: filters/block/block_test.go [hunk 4]
```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
...
+
+	spec := NewBlockFilter(1024)
+	args := []interface{}{"foo"}
+	fr := make(filters.Registry)
+	fr.Register(spec)
+	r := &eskip.Route{Filters: []*eskip.Filter{{Name: spec.Name(), Args: args}}, Backend: backend.URL}
+
+	proxy := proxytest.New(fr, r)
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:167]: defer proxy.Close()

File: filters/block/block_test.go [hunk 5]
```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
 
 		})
 	}
+}
+
+func TestMatcherErrorCases(t *testing.T) {
+	t.Run("maxBufferAbort", func(t *testing.T) {
+		r := &nonBlockingReader{initialContent: []byte("fppppppppp .class")}
+		toblockList := []toblockKeys{{str: []byte(".class")}}
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:89]: If toblockList is the same and not relevant for the test then maybe make it a reusable var in the TestMatcherErrorCases

File: filters/block/block_test.go [hunk 6]
```diff
@@ -69,7 +81,145 @@ func TestBlock(t *testing.T) {
...
+		bmb := newMatcher(r, toblockList, 5, maxBufferAbort)
+		p := make([]byte, len(r.initialContent))
+		_, err := bmb.Read(p)
+		if err != ErrMatcherBufferFull {
+			t.Errorf("Failed to get expected error %v, got: %v", ErrMatcherBufferFull, err)
+		}
+	})
+	t.Run("maxBuffer", func(t *testing.T) {
```
Comment (AlexanderYastrebov) [inline filters/block/block_test.go:97]: Missing newline before?

File: filters/block/block_test.go (hunk 1 above)
Comment (szuecs) [reply filters/block/block_test.go:117]: fixed

File: filters/block/block_test.go (hunk 2 above)
Comment (szuecs) [reply filters/block/block_test.go:131]: ah no thanks

File: filters/block/block_test.go (hunk 3 above)
Comment (szuecs) [reply filters/block/block_test.go:159]: fixed

File: filters/block/block_test.go (hunk 4 above)
Comment (szuecs) [reply filters/block/block_test.go:167]: fixed

File: filters/block/block_test.go (hunk 6 above)
Comment (szuecs) [reply filters/block/block_test.go:97]: fixed

File: filters/block/block_test.go (hunk 5 above)
Comment (szuecs) [reply filters/block/block_test.go:89]: fixed

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by MustafaSaber** [comment]
👍

**Comment by szuecs** [comment]
:+1:

**Pull request merged**